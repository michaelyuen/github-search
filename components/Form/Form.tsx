import { ReactNode } from "react";
import { FormContainer } from "./styles";
import { ComponentProps } from "../types";

export interface FormProps extends ComponentProps {
  children: ReactNode;
  name: string;
  noValidate?: boolean;
  onSubmit?(event: React.FormEvent<HTMLFormElement>): void;
}

export const Form: React.FC<FormProps> = ({ children, ...rest }) => {
  return (
    <FormContainer noValidate {...rest}>
      {children}
    </FormContainer>
  );
};
