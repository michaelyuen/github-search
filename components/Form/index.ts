export * from "./Form";
export * from "./styles";
