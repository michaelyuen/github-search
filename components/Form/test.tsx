import { render, screen } from "@testing-library/react";
import { axe } from "jest-axe";
import { Form } from "./Form";

describe("Form", () => {
  test("renders", () => {
    render(
      <Form aria-label="mock aria label" name="Mock Name">
        0
      </Form>
    );
    expect(screen.getByLabelText("mock aria label")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(
      <Form aria-label="mock aria label" name="Mock Name">
        0
      </Form>
    );
    expect(container).toMatchSnapshot();
  });

  test("has no a11y violations", async () => {
    const { container } = render(
      <Form aria-label="mock aria label" name="Mock Name">
        0
      </Form>
    );
    expect(await axe(container)).toHaveNoViolations();
  });
});
