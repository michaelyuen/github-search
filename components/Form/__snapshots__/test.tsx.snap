// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot 1`] = `
<div>
  <form
    aria-label="mock aria label"
    class="sc-bcXHqe"
    name="Mock Name"
    novalidate=""
  >
    0
  </form>
</div>
`;
